packages/core/src/dispatcher.ts

```diff
@@ -63,71 +63,39 @@ export default class Dispatcher extends EventEmitter {
   async init(): Promise<void> {
     this._peerId = this._peerId || (await this._ipfs.id()).id
     await this._subscribe()
-    !TESTING && this._resubscribeOnDisconnect()
+    if (!TESTING) {
+      this._resubscribe()
+    }
   }
 
   /**
-   * Subscribes IPFS pubsub to `this.topic` and logs an `subscribe` event.
+   * Subscribes IPFS pubsub to `this.topic` and logs a `subscribe` event.
+   *
+   * Logs error if subscribe fails.
    */
   async _subscribe(): Promise<void> {
-    if (this._isSubscribed) {
-      console.log(`Pubsub subscribe skipped. Already subscribed to topic ${this.topic}`)
-    } else {
-      try {
-        await this._ipfs.pubsub.subscribe(
-          this.topic,
-          this.handleMessage.bind(this),
-          {timeout: !TESTING && IPFS_GET_TIMEOUT}
-        )
-
-        const { isSubscribed, error } = await this._confirmIsSubscribed()
-        if (isSubscribed) {
-          this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
-        } else if (error) {
-          throw error
-        } else {
-          throw new Error(`Pubsub subscribe failed for topic ${this.topic}`)
-        }
-      } catch (error) {
-        console.error(error)
-      }
+    try {
+      await this._ipfs.pubsub.subscribe(
+        this.topic,
+        this._messageHandler,
+        {timeout: TESTING ? null : IPFS_GET_TIMEOUT}
+      )
+      this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
+    } catch (error) {
+      // TODO: use logger
+      console.error(error)
```

Comment:
Can we check the error to see if it's an "already subscribed" error and not log it, or log it at the debug log level?  Otherwise users might get concerned seeing lots of errors in their console that are actually expected

Reply:
don't want to tie logic to the error message in case it changes and there isn't a code produced. seems ok to do it this way for now and this is temporary anyway

Reply:
do we know *how* temporary?  I'd rather depend on the error message and deal with it if the message changes and that causes errors to slip out, rather than definitely have spurious errors filling our logs and causing concern among our users during their very first interaction with running a ceramic node

Reply:
can revisit this when we add the new logger

Reply:
how will the new logger help?  And why is tying logic to the error message so bad, other than being brittle?  I feel pretty strongly that we shouldn't launch Clay with the log filling with errors that aren't real errors.  I think that will look pretty poor to our community

Reply:
oh, I just saw that you did add an if-statement checking the log message, but we still log the error unconditionally.  I think the `console.error` line needs to be moved into an else-block

Reply:
updated